recipes/catch2/3.x.x/conanfile.py

```diff
@@ -0,0 +1,126 @@
+import os
```

Comment:
Can we re-use the same recipe for 2.x and 3.x? I didn't but they look pretty similar.

This is funny, in the past we Catch project (1.x), then they release version 2.x and renamed the project to Catch2. Now, they released the version 3.x but the names persists the same 😅

Reply:
This is a difficult issue for me to judge.

The first PR was 2.x.x with 3.0.1 support in it.
https://github.com/conan-io/conan-center-index/pull/10951

However, 3.0.1 has the following major changes for extensibility, 
and I thought it would be better to separate 3.x.x for future feature additions.

- Abolished single header and prepared headers for each function.
- Abolish "header only" and always link libraries

This is a problem about "How much change will separate the folders?" 
I think this is a matter of CCI policy, so I will follow your opinion.

Reply:
> This is funny, in the past we Catch project (1.x), then they release version 2.x and renamed the project to Catch2. Now, they released the version 3.x but the names persists the same :sweat_smile:

I agree with you.
I still wish they had released it as Catch3.

Reply:
Okay, so let's keep them separated. Major versions, sometimes, they are equal in terms of recipe, that's why I asked. Thank you for sharing the details.